### PR TITLE
Fix {enter} press in RTE

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

The current bug is happening when enter and a modifier key are pressed. The RTE is sending a message when shift/alt/meta/ctrl + enter is pressed instead to send only when enter is pressed (according to `ctrlEnterToSend` settings).

The fix it's a big ugly but I need to check if only the `enter` key is pressed.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix {enter} press in RTE ([\#9927](https://github.com/matrix-org/matrix-react-sdk/pull/9927)). Contributed by @florianduros.

### CODE REVIEW

File: src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts [hunk 1]
```diff
@@ -29,9 +29,14 @@ export function useInputEventProcessor(onSend: () => void): (event: WysiwygEvent
 
             const isKeyboardEvent = event instanceof KeyboardEvent;
             const isEnterPress =
-                !isCtrlEnter && (isKeyboardEvent ? event.key === "Enter" : event.inputType === "insertParagraph");
+                !isCtrlEnter &&
+                (isKeyboardEvent
+                    ? // Ugly but here we need to send the message only if Enter is pressed
+                      // And we need to stop the event propagation on enter to avoid the composer to grow
+                      event.key === "Enter" && !event.shiftKey && !event.ctrlKey && !event.metaKey && !event.altKey
+                    : event.inputType === "insertParagraph");
```
Comment (dbkr) [inline src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts:37]: Mm, this is making my head hurt somewhat. Use let & some separate if statements to conditionally change it? Or if it must be functional, define `isEnterPress` as the result of an anonymous function?

**Review by dbkr — COMMENTED** [review]
Also some more detail in the PR title could be useful, otherwise the tests are my best guess at what this actually does.

File: src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts (hunk 1 above)
Comment (florianduros) [reply src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts]: @dbkr I added some extra informations in the PR title and I have split the operation in a dedicated function in multiple booleans.

**Review by dbkr — APPROVED** [review]
thanks - lgtm

**Pull request merged**